include/mbgl/util/thread.hpp

```diff
@@ -66,6 +75,8 @@ class Thread : public Scheduler {
         }
 
         std::promise<void> joinable;
+        
+        running.wait();
 
         // Kill the actor, so we don't get more
```

Comment:
Seems like we should be able to get rid of this and rely solely on `loop->stop()`. Actor activation should be encapsulated in an RAII type that deactivates on destruction and can be on the thread's stack. Also needs investigation why `joinable` was needed.

Then, once we don't need `object.reset()`, it doesn't need to be a `unique_ptr`.

Reply:
Maybe we can encapsulate the two-phase construction in a pair of objects: `AspiringActor` (lives as `Thread` member data) and `EstablishedActor` (lives on the thread's stack).

Reply:
Using a separate type for the parent thread was how I started out (albeit not with nearly such clever naming ;-), but it started feeling overcomplicated.  But now that this approach is in place, yeah, lemme try an `AspiringActor`/`EstablishedActor` variation.

Before experimenting a bit in code, the thing I'm unsure of in such a model is where the `Object` (or rather, the `aligned_storage_t<sizeof(Object)>`) member belongs.  If it belongs to `AspiringActor`, then we'd have to use the Aspiring/Established pair (rather than just Established) even for non-`Thread<Object>` uses of Actor, which seems unfortunate.  But it can't belong to `EstablishedActor`, because then `AspiringActor` won't have an `Object*` for creating messages (right?).